### DISCUSSION TITLE
Question: what happens if the Redis gets full

### DISCUSSION THREAD

**Discussion by baspeti** [OptimalBits/bull#2174] [Q&A]
I read at this [issue](https://github.com/OptimalBits/bull/issues/1899#issuecomment-896250863), that `noeviction` should be set as Redis eviction policy. (And the same developer tells `noeviction` must be used with bull on the `taskforcesh` version of this queue.)

In this case, what would happen if the Redis gets full and `noeviction` is set? Would the queue crash or hang, because it might not be able to maintain its internal Redis keys? Or could the queue go on, especially if we stopped feeding the queue with jobs?

(Bonus question: what is the relationship between the OptimalBits and taskforcesh version of bull?)

**Comment by manast** [comment]
When redis gets full thing among other things you will not be able to add more jobs. There is nothing special about this, as maintainer of your redis instance you must make sure that it does not get full, set a warning when it reaches 80% or so.

**Reply by baspeti** [reply]
Thanks, yes, of course I will try everything to avoid Redis getting full, that's clear. The thing that I wanted to ask, what would happen if it got full by accident? That's clear that I would not be able add more jobs, I wanted to know how Bull would behave under this extreme situation. So if the Redis gets full by accident, and I stop adding more jobs, is Bull expected to "run existing jobs out" smoothly, or can it get wedged or hanged because it would not be able to maintain its internal Redis keys? What would be the expected behavior of Bull?

**Reply by manast** [reply]
You will get an undefined behaviour as in most DBs, because when memory is almost full, even deleting something from the database can fail.
